backend/query/test/items.e2e-spec.ts

```diff
@@ -1,42 +1,185 @@
-// import { INestApplication } from '@nestjs/common';
+import { HttpStatus, INestApplication, ValidationPipe } from '@nestjs/common';
+import { Model } from 'mongoose';
+import * as request from 'supertest';
+import { ItemsModule } from '../src/items/items.module';
+import { Item } from '../src/models/item.model';
+import { ItemsByTag } from '../src/models/items-by-tag.model';
+import { Tag } from '../src/models/tag.model';
+import {
+  initializeMockModule,
+  teardownMockDataSource,
+} from './helpers/MongoMemoryHelpers';
+
+import { ConfigModule } from '@nestjs/config';
+import { Test, TestingModule } from '@nestjs/testing';
+import { items, itemsWithDates, matchingItemNames } from './mocks/items';
+import { itemsByTags } from './mocks/itemsbytags';
+import { tags } from './mocks/tags';
 
 describe('QueryController (e2e)', () => {
-  // let app: INestApplication;
-  // let itemRepository: any;
+  let app: INestApplication;
+  let itemModel: Model<Item>;
+  let tagModel: Model<Tag>;
+  let itemsByTagModel: Model<ItemsByTag>;
+  let server: any; // Has to be any because of supertest not having a type for it either
+  jest.setTimeout(10000);
 
   beforeAll(async () => {
-    // const dataSource = await initializeMockDataSource();
-    // const moduleFixture: TestingModule = await Test.createTestingModule({
-    //   imports: [TypegooseModule.forRoot(''), CommandsModule],
-    // }).compile();
-    // app = moduleFixture.createNestApplication();
-    // app.useGlobalPipes(new ValidationPipe({ whitelist: true }));
-    // await app.init();
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [
+        ConfigModule.forRoot({ isGlobal: true }),
```

Comment:
Does this require .env file in the rootdir? If no add the `ignoreEnvFile` option to decouple it from .env file changes